bundle/IbexaSeoToolkitBundle.php

```diff
@@ -25,6 +26,9 @@ public function build(ContainerBuilder $container): void
     {
         parent::build($container);
 
+        $container->registerForAutoconfiguration(AnalyzerInterface::class)
```

Comment:
The `Codein\IbexaSeoToolkit\Analysis\AnalyzerInterface` is already registered for autoconfiguration through the yaml configuration https://github.com/Codein-Labs/ibexa-seo-toolkit/blob/master/bundle/Resources/config/services.yaml#L9
Normally, your custom analyzer should be resolved without doing any extra config. Could you please take a quick look !

Reply:
The autoconfiguration declared in the services.yaml applies only for the classes inside the bundle. 
Without registering the interface, `debug:container` gives me the following result : 

```
Information for Service "Codein\IbexaSeoToolkit\Analysis\Analyzers\InternalLinksAnalyzer"
=========================================================================================

 Class InternalLinksAnalyzer.

 ---------------- ----------------------------------------------------------------- 
  Option           Value                                                            
 ---------------- ----------------------------------------------------------------- 
  Service ID       Codein\IbexaSeoToolkit\Analysis\Analyzers\InternalLinksAnalyzer  
  Class            Codein\IbexaSeoToolkit\Analysis\Analyzers\InternalLinksAnalyzer  
  Tags             codein_ibexa_seo_toolkit.seo_analyzer                            
  Public           yes                                                              
  Synthetic        no                                                               
  Lazy             no                                                               
  Shared           yes                                                              
  Abstract         no                                                               
  Autowired        yes                                                              
  Autoconfigured   yes                                                              
 ---------------- ----------------------------------------------------------------- 

Information for Service "App\SeoAnalyzer\InternalLinksAnalyzer"
===============================================================

 ---------------- --------------------------------------- 
  Option           Value                                  
 ---------------- --------------------------------------- 
  Service ID       App\SeoAnalyzer\InternalLinksAnalyzer  
  Class            App\SeoAnalyzer\InternalLinksAnalyzer  
  Tags             -                                      
  Public           no                                     
  Synthetic        no                                     
  Lazy             no                                     
  Shared           yes                                    
  Abstract         no                                     
  Autowired        yes                                    
  Autoconfigured   yes                                    
 ---------------- --------------------------------------- 
```

In the bundle, the service is taggeg the right way. Outside, it's not tagged

I haven't done many research on this topic but noticed this behaviour with other bundles I worked on.
I think it's the right way to tag the third parties analyzers. The main reason is that the registration through the bundle declaration will be taken taken in account as soon as the bundle is instanciated. And that will happend before the bundle's services.yaml applies.

Reply:
Good catch! The _defaults and _instances keys are scoped per files, so they are only applies to services inside the bundle. This makes sense, global configuration should not be impacted but bundle configuration. You can remove these 2 useless lines and merge your pull request.
https://github.com/Codein-Labs/ibexa-seo-toolkit/blob/17823071e8ada0c31e0dd306d31629e877b82c53/bundle/Resources/config/services.yaml#L9-L10
By the way, I think it's more appropriate to centralize the logic of registering interfaces for "autoconfiguration" at the end of the `IbexaSeoToolkitExtension::build` function, as since DI extensions have already access to service configuration.

Reply:
Done !